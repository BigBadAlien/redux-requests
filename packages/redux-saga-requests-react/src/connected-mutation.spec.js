import renderer from 'react-test-renderer';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import ConnectedMutation from './connected-mutation';

const mockStore = configureStore();

describe('ConnectedMutation', () => {
  it('maps requestSelector to mutation with type prop', () => {
    const mutationType = 'MUTATION';
    const component = renderer.create(
      <Provider
        store={mockStore({
          request: {
            mutations: { [mutationType]: { pending: 1, error: 'error' } },
          },
        })}
      >
        <ConnectedMutation
          requestSelector={state => state.request}
          type={mutationType}
        >
          {({ loading, error }) => (
            <div>
              {loading && 'loading'}
              {error}
            </div>
          )}
        </ConnectedMutation>
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('maps requestSelector to mutation with type prop as action creator', () => {
    const mutationAction = () => {};
    mutationAction.toString = () => 'MUTATION';

    const component = renderer.create(
      <Provider
        store={mockStore({
          request: {
            mutations: { MUTATION: { pending: 1, error: 'error' } },
          },
        })}
      >
        <ConnectedMutation
          requestSelector={state => state.request}
          type={mutationAction}
        >
          {({ loading, error }) => (
            <div>
              {loading && 'loading'}
              {error}
            </div>
          )}
        </ConnectedMutation>
      </Provider>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
