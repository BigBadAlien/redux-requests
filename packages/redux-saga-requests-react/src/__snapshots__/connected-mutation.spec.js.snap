// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedMutation maps requestSelector to mutation with type prop 1`] = `
<div>
  loading
  error
</div>
`;

exports[`ConnectedMutation maps requestSelector to mutation with type prop as action creator 1`] = `
<div>
  loading
  error
</div>
`;
